{"name": "django-filters", "cve_id": "GHSA-x7gm-rfgv-w973", "publish_date": "2020-09-28T19:05:29Z", "cwe_id": "CWE-400", "owner": "Alex Gaynor", "affected_versions": [{"version_value": "2.4.0", "operator": "<"}], "references": ["https://github.com/carltongibson/django-filter/security/advisories/GHSA-x7gm-rfgv-w973", "https://nvd.nist.gov/vuln/detail/CVE-2020-15225", "https://github.com/advisories/GHSA-x7gm-rfgv-w973"], "description": "### Impact\n\nAutomatically generated `NumberFilter` instances, whose value was later converted to an integer, were subject to potential DoS from maliciously input using exponential format with sufficiently large exponents. \n\n### Patches\n\nVersion 2.4.0+ applies a `MaxValueValidator` with a a default `limit_value` of 1e50 to the form field used by `NumberFilter` instances. \n\nIn addition, `NumberFilter` implements the new `get_max_validator()` which should return a configured validator instance to customise the limit, or else `None` to disable the additional validation. \n\n### Workarounds\n\nUsers may manually apply an equivalent validator if they are not able to upgrade.  \n\n### For more information\nIf you have any questions or comments about this advisory:\n* Open an issue in [the django-filter repo](https://github.com/carltongibson/django-filter)\n\nThanks to Marcin Waraksa for the report. ", "latest_version": "2.4.0", "language": "python", "license": "BSD-3-Clause", "repository_summary": "A generic system for filtering Django QuerySets based on user selections", "repo_url": "https://github.com/carltongibson/django-filter", "impact": {"cvss3_score": 6.0, "cvss3_severity": "MEDIUM", "vector_string": "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H", "av": "NETWORK", "ac": "LOW", "pr": "LOW", "ui": "NONE", "s": "UNCHANGED", "c": "NONE", "i": "NONE", "a": "HIGH", "impact_score": null, "exploitability_score": null}, "ochrona_severity_score": 6}